geode-core/src/main/java/org/apache/geode/internal/DeployedJar.java

```diff
@@ -366,14 +368,23 @@ private Function newFunction(final Class<Function> clazz, final boolean errorOnN
   }
 
   /**
-   * @return the unversioned name of this jar file, e.g. myJar.jar
+   * Get this jar's artifact ID, which is the part of the jar file name that precedes the version
+   * information.
+   *
+   * @return the artifact ID for this jar
    */
-  public String getJarName() {
-    return this.jarName;
+  public String getArtifactId() {
+    return this.artifactId;
   }
 
-  public String getFileName() {
-    return this.file.getName();
+  /**
+   * @return the filename as user deployed, i.e remove the sequence number
+   */
+  public String getDeployedFileName() {
+    String fileName = this.file.getName();
+    Matcher matcher = SEQUENCED_VERSION_SCHEME.matcher(fileName);
+    matcher.find();
+    return matcher.group("artifact") + ".jar";
```

Comment:
I find the group name "artifact" confusing here because this is not the same as what getArtifactId returns. In this case it is just the sequence number we remove. Would a better name for this group be "deployed"?